app.rb

```diff
@@ -22,6 +23,99 @@ def list_books
     end
   end
 
+  def save_data
+    book_list = []
+    rental_list = []
+    persons_list = []
+    @books.each do |book|
+      book_obj = {
+        title: book.title,
+        author: book.author
+      }
+      book_list << book_obj
+    end
+    File.write('./memory/books.json', JSON.pretty_generate(book_list))
+
+    @rentals.each do |rental|
+      rental_obj = {
+        date: rental.date,
+        book: rental.book.title,
+        person: rental.person.name
+      }
+      rental_list << rental_obj
+    end
+    File.write('./memory/rentals.json', JSON.pretty_generate(rental_list))
+
+    @persons.each do |person|
+      persons_obj = if person.instance_of?(Teacher)
+                      {
+                        class_name: person.class,
+                        age: person.age,
+                        name: person.name,
+                        specialization: person.specialization
+                      }
+                    else
+                      {
+                        class_name: person.class,
+                        age: person.age,
+                        name: person.name,
+                        classroom: person.classroom,
+                        parent_permission: person.parent_permission
+                      }
+                    end
+      persons_list << persons_obj
+    end
+    File.write('./memory/persons.json', JSON.pretty_generate(persons_list))
+  end
```

Comment:
- [ ] I noticed that when I exit the application, I get the error as shown in the screenshot below and the data is not persisted in the _**JSON**_ files

![image](https://user-images.githubusercontent.com/77106962/225461612-83503dfd-a2e3-40e9-92a2-2d4142cbdcfc.png)

To solve this, you would need to create the directory `memory` before you can create or write in JSON files. Also, the logic for creating the directory `memory` should be that if it exists, it shouldn't be created. 👍 😎

Reply:
Thanks for the observation and advice. We'll fix it!